### ISSUE TITLE
FSL Version 6.0.6.4

### ISSUE THREAD

**Issue by dnkennedy** [ReproNim/neurodocker#531]
Hi, I believe that the documentation says FSL 6.0.6.4 is supported (https://www.repronim.org/neurodocker/user_guide/cli.html suggests that
--fsl KEY=VALUE              Add fsl
                                method=[binaries]
                                options for method=binaries
                                - version [required]
                                    version=[6.0.6.4|6.0.6.3|6.0.6.2|6.0.6.1|6.0.6|6.0.5.2|6.0.5.1|6.0.5|6.0.4|6.0.3|6.0.2|6.0.1|6.0.0|5.0.9|5.0.8|5.0.11|5.0.10]
                                - install_path [default: /opt/fsl-{{ self.version }}]
                                - exclude_paths [default: ]
                            **Note**: FSL is non-free. If you are considering commercial use of FSL, please consult the relevant license(s).

When running the following command:

$ docker run -i --rm repronim/neurodocker:latest  generate docker \
    --pkg-manager apt \
    --base-image debian:buster-slim \
    --fsl version=6.0.6.4 \
> fsl6064.Dockerfile

I get:
FSL is non-free. If you are considering commercial use of FSL, please consult the relevant license(s). Proceed? [y/N]:y
Traceback (most recent call last):
  File "/opt/neurodocker/neurodocker/reproenv/renderers.py", line 213, in from_dict
    renderer.add_registered_template(method_or_template, **kwds)
  File "/opt/neurodocker/neurodocker/reproenv/renderers.py", line 366, in add_registered_template
    self.add_template(template=template, method=method)
  File "/opt/neurodocker/neurodocker/reproenv/renderers.py", line 264, in add_template
    template_method.validate_kwds()
  File "/opt/neurodocker/neurodocker/reproenv/template.py", line 164, in validate_kwds
    raise TemplateKeywordArgumentError(
neurodocker.reproenv.exceptions.TemplateKeywordArgumentError: Unknown version '6.0.6.4'. Allowed versions are '6.0.5.1', '6.0.0', '6.0.1', '5.0.8', '6.0.5', '5.0.11', '6.0.2', '6.0.4', '5.0.10', '6.0.3', '5.0.9'.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/neurodocker", line 33, in <module>
    sys.exit(load_entry_point('neurodocker', 'console_scripts', 'neurodocker')())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/neurodocker/neurodocker/cli/generate.py", line 441, in docker
    _base_generate(
  File "/opt/neurodocker/neurodocker/cli/generate.py", line 418, in _base_generate
    r = renderer.from_dict(renderer_dict)
  File "/opt/neurodocker/neurodocker/reproenv/renderers.py", line 215, in from_dict
    raise RendererError(
neurodocker.reproenv.exceptions.RendererError: Error on template 'fsl'. Please see above for more information.

Key point being "Unknown version '6.0.6.4'. Allowed versions are '6.0.5.1', '6.0.0', '6.0.1', '5.0.8', '6.0.5', '5.0.11', '6.0.2', '6.0.4', '5.0.10', '6.0.3', '5.0.9'." whereas the allowed version do not seem to match the documentation...

my neuodocker version is:
docker run --rm repronim/neurodocker:latest --version
neurodocker version 0.9.4

which I believe to be latest...

Running on Macos 11.7.6
Docker(desktop) Version 4.18.0 (104112)

Let me know what I'm missing, thanks!

**Comment by kaczmarj** [comment]
fsl 6.0.6.4 was added in https://github.com/ReproNim/neurodocker/pull/527 but we have not cut a new release since then. @djarecka how do we cut a new release now? i know that it's partially automated but i don't know how to trigger it

**Comment by satra** [comment]
@kaczmarj - take any open PR and add release + major/minor/patch (as relevant to the release) tag to it. once that PR is merged it will auto release.

**Comment by dnkennedy** [comment]
Thanks, all. I will await, patiently, the next tagged release.  Is there a temporary workaround for using the untagged version that supports this version of FSL?

**Comment by kaczmarj** [comment]
@dnkennedy i have released a new version 0.9.5, which includes fsl 6.0.6.4.

```
docker run repronim/neurodocker:0.9.5
```

for the future, assuming the features you need are in the master branch but not yet released, you can use the docker image tagged `master`.

```
docker run repronim/neurodocker:master
```

**Comment by dnkennedy** [comment]
Thank you soo much, @kaczmarj !

**Comment by dnkennedy** [comment]
Seems to work well.  Is there a good way to double check the fsl version in the image?

I can do 
flirt -version

but that provides a pretty generic "FLIRT version 6.0" on the docker which is the same as I get for my FSL 6.0.3 image...

**Comment by dnkennedy** [comment]
Ah, cat $FSLDIR/etc/fslversion. I'll see if I can figure out how to pass that inquiry to the container...

**Comment by dnkennedy** [comment]
Ack...

$ docker run --rm -it dnk/fsl:6.0.6.4 more '$FSLDIR/etc/fslversion'
more: stat of $FSLDIR/etc/fslversion failed: No such file or directory

How to pass the string $FSLDIR not expanded in my docker call, but have it expand within the container???

**Comment by dnkennedy** [comment]
OK, answered my own question, thanks for being out there and listening:

$ docker run --rm -it dnk/fsl:6.0.6.4 sh -c 'more $FSLDIR/etc/fslversion'
6.0.6.4


I deem this issue closed! Thanks all...

**Comment by kaczmarj** [comment]
Excellent! Glad you figured this out. And thanks for the bug report :)Best,JakubOn May 12, 2023, at 14:00, David Kennedy ***@***.***> wrote:﻿
Closed #531 as completed.
